simd/unit_tests/include/SIMDTesting_Ops.hpp

```diff
@@ -80,7 +80,7 @@ class absolutes {
   template <typename T>
   auto on_host(T const& a) const {
     if constexpr (std::is_signed_v<typename T::value_type>) {
-#if defined(KOKKOS_ENABLE_DEPRECATED_CODE_4) && !defined(KOKKOS_COMPILER_NVCC)
+#if defined(KOKKOS_ENABLE_DEPRECATED_CODE_4)
```

Comment:
What about https://github.com/kokkos/kokkos/pull/6485/files#diff-66339256bc16b14bde61943b2f7e184363640137e9a9b03ccf93c7f1a89e3fa9L98-L102 ?
Should we revert it as well?

Reply:
Reverting that change would make us call `__host__` functions from a `__host__ __device__` function (for `host`-only types) and we would potentially run into the same error since we would be trying to use vector types in a `__host__ __device__` function.